src/qthotkey/plugin.h

```diff
@@ -0,0 +1,53 @@
+#ifndef _PLUGIN_H_INCLUDED_
+#define _PLUGIN_H_INCLUDED_
+
+#include <QtCore/QList>
+
+namespace GlobalHotkeys {
+
+enum EVENT {
+	EVENT_PREV_TRACK = 0,
+	EVENT_PLAY,
+	EVENT_PAUSE,
+	EVENT_STOP,
+	EVENT_NEXT_TRACK,
+
+	EVENT_FORWARD,
+	EVENT_BACKWARD,
+	EVENT_MUTE,
+	EVENT_VOL_UP,
+	EVENT_VOL_DOWN,
+	EVENT_JUMP_TO_FILE,
+	EVENT_TOGGLE_WIN,
+	EVENT_SHOW_AOSD,
+
+	EVENT_TOGGLE_REPEAT,
+	EVENT_TOGGLE_SHUFFLE,
+	EVENT_TOGGLE_STOP,
+
+	EVENT_RAISE,
+
+	EVENT_MAX
+};
+
+struct HotkeyConfiguration {
```

Comment:
Most likely this could be done with a `QMap` or similar?

Reply:
It's a design approach I've taken from gtk global hotkeys plugin. The preferences widget doesn't have fixed slot for each action and instead it allows to add unlimited amount of hotkey combinations. It means, for one action it's possible to set a few different hotkey combinations.

Unless this design needs to be reworked and restricted to one hotkey combination per possible action, I'd prefer to not change it.